apps/merkle_patricia_tree/lib/merkle_patricia_tree/storage_behaviour.ex

```diff
@@ -0,0 +1,28 @@
+defmodule MerklePatriciaTree.StorageBehaviour do
+  @moduledoc """
+  Defines functions for fetching and updating nodes.
+  """
+
+  alias MerklePatriciaTree.Trie
+  alias MerklePatriciaTree.Trie.Node
+
+  @type t :: struct()
+
+  @callback fetch_node(t) :: Node.trie_node()
+
+  @callback put_node(t, node) :: nil | binary()
+
+  @callback remove_key(t, Trie.key()) :: t
+
+  @callback update_key(t(), Trie.key(), ExRLP.t() | nil) :: t
+
+  @callback get_key(t(), Trie.key()) :: t()
+
+  @callback into(binary(), t) :: t
+
+  @callback store(t) :: t
```

Comment:
Should this be a `@callback`? The cache implementation doesn't seem to implement this, so I would say that we should probably move this out of the behaviour and just let it be a function in the `Trie` module.

Reply:
`CachingTrie` implements `store` https://github.com/mana-ethereum/mana/blob/9704ade786d567e34e4e703532d706a2ae6dedf7/apps/merkle_patricia_tree/lib/merkle_patricia_tree/caching_trie.ex#L89

Reply:
My bad. I missed that.